### PR TITLE
Increase GC timeout for RCs bigger than 40k

### PR DESCRIPTION
**Pull request by gmarek**
cc @caesarxuchao

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/32062)

### CODE REVIEW

File: test/e2e/framework/util.go [hunk 1]
```diff
@@ -163,6 +163,9 @@ const (
 
 	// Number of times we want to retry Updates in case of conflict
 	UpdateRetries = 5
+
+	// number of objects that gc can delete in a second
+	gcThroughput = 15
```
Comment (caesarxuchao) [inline test/e2e/framework/util.go]: Not sure where "15" comes from. GC throughput is QPS/2 since it sends 2 requests to delete an object. The maths worked last time I checked the 500-gce density test :)

File: test/e2e/framework/util.go [hunk 2]
```diff
@@ -3355,16 +3358,15 @@ func DeleteRCAndWaitForGC(c *client.Client, ns, name string) error {
 	switch {
 	case rc.Spec.Replicas < 100:
 		interval = 10 * time.Millisecond
```
Comment (wojtek-t) [inline test/e2e/framework/util.go]: 10ms here is pretty aggresive - I would do 100ms

File: test/e2e/framework/util.go (hunk 1 above)
Comment (wojtek-t) [inline test/e2e/framework/util.go]: @caesarxuchao Did you use default QPS limits in density? In Jenkins we have them increased, which means that it's more likely to pass in Jenkins.
BTW - it seems that indeed it is passing in 500-gce density, but it's not passing in 2000-gce density.

@gmarek - I also don't know where 15 comes from... :) I would suggest 10 or sth like (with some explanatory comment like what @caesarxuchao wrote above).

Comment (gmarek) [inline test/e2e/framework/util.go]: It came from previous settings - 10k pods -> 10 minutes, which gives 16.6 pods/s, so I rounded it down to 15:) I can do 10 and change the limit for 10k as well.

File: test/e2e/framework/util.go (hunk 2 above)
Comment (gmarek) [inline test/e2e/framework/util.go]: Done.

File: test/e2e/framework/util.go [hunk 3]
```diff
@@ -163,6 +163,9 @@ const (
 
 	// Number of times we want to retry Updates in case of conflict
 	UpdateRetries = 5
+
+	// number of objects that gc can delete in a second
```
Comment (wojtek-t) [inline test/e2e/framework/util.go]: Can you please add a comment with what @caesarxuchao mostly wrote (that we have two operations per delete) (to show that this 10 is not a random number).

**Comment by wojtek-t** [comment]
Minor thing - other than that lgtm.

File: test/e2e/framework/util.go (hunk 3 above)
Comment (gmarek) [inline test/e2e/framework/util.go]: Done.

**Comment by caesarxuchao** [comment]
> Did you use default QPS limits in density? In Jenkins we have them increased, which means that it's more likely to pass in Jenkins.

GC respects the QPS set by the test. GC uses the same client config used by other controllers, GC is not special.

>  it seems that indeed it is passing in 500-gce density, but it's not passing in 2000-gce density.

It seems It's passing 2000-gce as well? `kubernetes-kubemark-gce-scale` is the 2000-gce, right? It took GC 20mins to delete the 60,000 pods there. It's as expected, QPS in the test is 100, so 60,000/(100/2)/60=20 minutes. (And the duration is smaller than the timeout before this PR, which is 40 mins)

Which test is this PR trying to fix?

**Comment by wojtek-t** [comment]
>
It seems It's passing 2000-gce as well? kubernetes-kubemark-gce-scale is the 2000-gce, right? It took GC 20mins to delete the 60,000 pods there. It's as expected, QPS in the test is 100, so 60,000/(100/2)/60=20 minutes. (And the duration is smaller than the timeout before this PR, which is 40 mins)
Which test is this PR trying to fix?

Yes - kubemark-scale is passing, because it has 5x higher QPS limits than default ones. This isn't/wasn't the case in case of "kubernetes-e2e-gce-enormous-cluster" suite, where it took more than an hour to delete all pods.

In other words, default QPS limits are still 20, which means that GC by default won't delete more than 10pods/s.

**Comment by caesarxuchao** [comment]
I see. It will take 100 mins in that test. Thanks for the explanation.
